### ISSUE TITLE
Collaborators

### ISSUE THREAD

**Issue by icebob** [icebob/vue-form-generator#246]
## Current collaborators
@icebob
@lionel-bijaoui 
@dflock
@cristijora

## Rules/Rights for collaborators
- create/close/modify/organize issues
- create/merge/close PRs
- **Merge PRs only after @icebob approved!**
- Open a discussion issue (and mention other collaborators) before create a new feature PR (need to discuss all new features with the team)

**Comment by icebob** [comment]
Welcome @cristijora as our new collaborator (of course if he accept my invite :smile: )!

**Comment by cristijora** [comment]
Thanks!

**Comment by icebob** [comment]
Welcome @zoul0813 as our new collaborator!

**Comment by zoul0813** [comment]
Thanks @icebob

**Comment by lionel-bijaoui** [comment]
@icebob I would like to work on a v3 for VFG, can you create an branch for it ?

**Comment by icebob** [comment]
Yes. It's [done](https://github.com/icebob/vue-form-generator/tree/v3).

**Comment by lionel-bijaoui** [comment]
Thank you !

**Comment by icebob** [comment]
Hi guys,

I automated the NPM publishing. Travis publishing the tagged commits. It means every collaborator can publish a new version to NPM (without me) :)
There is a new npm command to bump version, build files, tagging, and pushing. 

**For patch version _(use after bug fixes)_**
```
$ npm run version patch
```

**For minor version _(use after new implemented features)_**
```
$ npm run version minor
```

**For major version _(use after breaking changes)_**
```
$ npm run version major
```

This executes the following tasks:
- bump version in `package.json` & `bower.json`
- `npm run build` so that dist files banner contains the new version number
- update `CHANGELOG.md`
- commit all changes files and push to Github
- create a new git tag & push to Github
- Travis will run publishing if tests successes

**Result on Travis CI (only on Node 8 test):**
![image](https://user-images.githubusercontent.com/306521/37205943-f8876328-2396-11e8-91de-04b023d8ed92.png)

**Technical details:**
It uses [bumped](https://github.com/bumped/bumped) to execute tasks & bump version.

>VFG v2.2.1 has been released with this command.

**Comment by zoul0813** [comment]
The npm publish done by Travis will only occur if we bump the version - correct?

**Comment by icebob** [comment]
Yes, but the `npm run version <type>` bumps the version. So no need manually.
Travis doesn't check the version. It triggered by pushed git tags.

**Comment by zoul0813** [comment]
Ok, just making sure Travis wasn't going to automatically publish every commit to master.  I assumed it wouldn't, but figured I'd ask just to double check.

**Comment by icebob** [comment]
No no.

**Comment by lionel-bijaoui** [comment]
What about merge ? Should we still wait for you approval @icebob ?

**Comment by icebob** [comment]
No, because I give the project to the team. But I recommend that at least one other collaborator review all PRs, right?

**Comment by lionel-bijaoui** [comment]
That's fine by me, I just wanted to be sure about that. I remember this being a major rule to follow 😄 

Maybe 2 other collaborator ?

**Comment by icebob** [comment]
Yes, but at that time I had enough free time to check all changes and issues. And now I haven't and I don't want to be the bottle-neck.

**Comment by icebob** [comment]
2 others is ok. But if no 2 other active collaborator after 2-3 days?

**Comment by lionel-bijaoui** [comment]
Yes, keep the thing rolling even if nobody answer.

So, if I create a PR : 
* I add everyone for review.
* If I get 2 validations in less than 3 days, I can merge.
* If I get 1 validation in 3 days or more, I can merge.
* If I get a rejection, I can't merge as long as I haven't satisfied the demands. Even if I have 2 validations.
* If there is a conflict about a PR, everyone have to vote and discuss it to find a solution.
* The majority win if all else failed.

Does it sound fair ?

**Comment by lionel-bijaoui** [comment]
Also, we should add a `ISSUE_TEMPLATE.md` to ask for a fiddle if this is a bug, expected vs observed behavior, etc.

**Comment by zoul0813** [comment]
I prefer the `@vue-form-generator/field-checklist` convention.  Prefixing field components with `field`, so as not to confuse them with other possible modules we may come up with (validation modules, template/theme/style modules, etc).  I prefer the kebab-case, personally.

As for the Fiddle, I think we need to update the current Fiddle referenced in the docs to use the latest VFG and Vue... and I agree, we should have something we can reference when requesting Fiddles for issues.

**Comment by icebob** [comment]
@lionel-bijaoui I think yes, you can create modules, but I don't know, it is my first org with other users :) So would be good to try it and if not working I will try to set the correct rights.
I prefer the `@vue-form-generator/field-checklist` type. Maybe in the future, we will have other types, like `validator-xxx`...etc

Yes, ISSUE template would be great. Could you make one?

**Comment by icebob** [comment]
@lionel-bijaoui what is your npmjs.com username?

**Comment by lionel-bijaoui** [comment]
@icebob I don't have an account, let me create one

**Comment by lionel-bijaoui** [comment]
About the naming convention, I forget that we decided `vfg-field-***` (eg. `@vue-form-generator/vfg-field-checklist`)
We have our first external field #419 !
